openmp/libomptarget/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -658,7 +664,8 @@ struct AMDGPUQueueTy {
     Packet->grid_size_x = NumBlocks * NumThreads;
     Packet->grid_size_y = 1;
     Packet->grid_size_z = 1;
-    Packet->private_segment_size = Kernel.getPrivateSize();
+    Packet->private_segment_size =
+        Kernel.usesDynamicStack() ? Kernel.getPrivateSize() : StackSize;
```

Comment:
Naming here is kind of confusing. For the most part "private size" and "stack size" are the same thing

Reply:
I can rename it to all be `private size`.

Reply:
I don't understand the control flow here.

Before, we called getPrivateSize() which retrieved N from somewhere, presumably kernel metadata.

Now, we only do that if useDynamicStack is true, otherwise it's taken from a new argument.

Condition the wrong way around?